File: backend/llm/app/model/NotificationCondition.py
from typing import Optional
from sqlalchemy import Integer, String, ForeignKey
from sqlalchemy.orm import relationship, Mapped, mapped_column
from app.database import Base

class NotificationCondition(Base):
    __tablename__ = "notification_condition"

    id: Mapped[int] = mapped_column("notification_condition_id", Integer, primary_key=True, autoincrement=True)
    target_price: Mapped[int] = mapped_column(String, nullable=False, comment="지정가")
    active: Mapped[str] = mapped_column(String(1), default="Y", comment="알림 조건 활성화 여부")
    member_id: Mapped[int] = mapped_column(Integer, ForeignKey("member.id"), nullable=False, comment="멤버 ID")

    # 가격알림용 임시 속성
    category: Mapped[Optional[str]] = mapped_column(String, nullable=True, comment="대분류")
    item: Mapped[Optional[str]] = mapped_column(String, nullable=True, comment="중분류")
    variety: Mapped[Optional[str]] = mapped_column(String, nullable=True, comment="하위 품종")
    region: Mapped[Optional[str]] = mapped_column(String, nullable=True, comment="지역")

    member = relationship("Member", back_populates="notification_conditions")

    def __repr__(self):
        return f"<NotificationCondition id={self.id} target_price={self.target_price}>"

File: backend/core/src/main/java/com/bombi/core/domain/notificationcondition/model/NotificationCondition.java
package com.bombi.core.domain.notificationcondition.model;

import static jakarta.persistence.FetchType.*;

import org.hibernate.annotations.Comment;

import com.bombi.core.domain.base.model.BaseEntity;
import com.bombi.core.domain.member.model.Member;
import com.bombi.core.domain.product.model.Product;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class NotificationCondition extends BaseEntity {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "notification_condition_id")
	private Long id;

	@Column(columnDefinition = "VARCHAR(50) NOT NULL")
	@Comment("지정가")
	private int targetPrice;

	@Column(columnDefinition = "VARCHAR(1) NOT NULL")
	@Comment("가격 방향성(U/D)")
	private int priceDirection;

	@Column(columnDefinition = "VARCHAR(1) NOT NULL")
	@Comment("알림 조건 활성화 여부")
	private String active;

	@ManyToOne(fetch = LAZY)
	@JoinColumn(name = "member_id")
	@Comment("멤버 ID")
	private Member member;

//	@ManyToOne(fetch = LAZY)
//	@JoinColumn(name = "product_id", columnDefinition = "BIGINT NOT NULL")
//	@Comment("작물 ID")
//	private Product product;

	// 가격알림용 임시속성
	private String category;
	private String item;
	private String variety;
	private String region;

	public boolean isActive() {
		return "T".equals(this.active);
	}
}
